lib/osf-components/addon/components/addons-service/addon-account-setup/component.ts

```diff
@@ -38,25 +40,32 @@ interface InputFieldObject {
 interface Args {
     onConnect: () => void;
     onReconnect: () => void;
-    reconnect?: boolean;
+    account?: AllAuthorizedAccountTypes;
     provider: ExternalStorageServiceModel; // Type?
     manager: AddonsServiceManagerComponent | UserAddonsManagerComponent;
-    onInput: (event: Event) => void;
 }
 
 export default class AddonAccountSetupComponent extends Component<Args> {
     @service intl!: IntlService;
     @service toast!: Toast;
 
     @tracked selectedRepo?: string;
-    @tracked otherRepo = '';
-    @tracked account?: AllAuthorizedAccountTypes;
+    @tracked otherRepo?: string;
+    @tracked url?: string;
+    @tracked newAccount?: AllAuthorizedAccountTypes;
     @tracked pendingOauth = false;
+    @tracked credentialsObject: AddonCredentialFields = {};
+    @tracked displayName = this.args.account?.displayName || '';
+    @tracked connectAccountError = false;
 
     get useOauth() {
         return [CredentialsFormat.OAUTH, CredentialsFormat.OAUTH2].includes(this.args.provider.credentialsFormat);
     }
 
+    get showUrlField() {
+        return this.args.provider.credentialsFormat === CredentialsFormat.URL_USERNAME_PASSWORD;
```

Comment:
Is there going to be a `URL_USERNAME_PASSWROD` credentials format from GV, or has that gone away since URL isn't a credential? @aaxelb

Reply:
yeah `URL_USERNAME_PASSWORD` is no longer since the url is not part of credentials -- ([these docs need a better docstring](https://addons.staging.osf.io/static/gravyvalet_code_docs/addon_service/common/credentials_formats.html#CredentialsFormats.USERNAME_PASSWORD) but the current credentials formats options are:
```
UNSPECIFIED
OAUTH2
ACCESS_KEY_SECRET_KEY
USERNAME_PASSWORD
PERSONAL_ACCESS_TOKEN
OAUTH1A
```

Reply:
So what is the proper way of specifying that we need a repo url for any given addon?

Reply:
i think the external-service's [configurableApiRoot](https://github.com/CenterForOpenScience/ember-osf-web/pull/2273/files#diff-9d8e5e59e7f153794afe9a0244c2d5bfa3741082521e8ae0e8312cbcae095b48R33) boolean, which might be:
```suggestion
        return this.args.provider.configurableApiRoot;
```
(if "provider" here means what i think it does)